src/internal/ppsload/pipeline.go

```diff
@@ -93,3 +116,32 @@ func createPipeline(pachClient *client.APIClient, repo *pfs.Repo, inputRepos []s
 	)
 	return errors.EnsureStack(err)
 }
+
+const stateFileName = "state"
+
+func serializeState(pachClient *client.APIClient, state *State) (string, error) {
```

Comment:
This code looks the same as serialize/deserialize in pfs but I guess it can't be cleanly extracted into a single function since `State` is defined differently in `pfs` and `pps`

Reply:
Yeah, we have the same issue with the tasks in the task service. I am very open to smarter ways to handle this.